libs/api/src/client_utils.ts

```diff
@@ -0,0 +1,41 @@
+import { safeParseJson } from '@votingworks/types';
+import * as z from 'zod';
+
+import { ErrorsResponse, OkResponse } from './base';
+
+/**
+ * A Fetch wrapper for API clients that accepts an API response schema for
+ * parsing and validating server responses.
+ */
+export async function fetchWithSchema<T extends OkResponse | ErrorsResponse>(
```

Comment:
I might check with @eventualbuddha to see if we have an existing pattern for passing errors from the backend. I haven't worked with this codebase at all, so I'm not sure off the top of my head. I poked around a little and found [this test case](https://github.com/votingworks/vxsuite/blob/b891f3dd43b688fe9d1b3e561e227d5670bf8dbd/frontends/election-manager/src/lib/backends/admin_backend.test.ts#L114-L132) that looks like it shows how an error gets passed from backend to frontend.

Also in terms of code organization, I'm not sure, but maybe putting it in the same place as `fetchJson` would make sense?

Reply:
oh, thanks for that test case - that makes me wonder if maybe the intention is to just always return HTTP-200, even when we're returning errors. that reduces the necessity of this util, although this would still give us the schema/type safety.

@eventualbuddha - should api-generated errors be returned with status 200?

Reply:
and re: co-locating with `fetchJson`, i initially started doing that and hit a circular reference issue when trying to reference the response types from `@votingworks/api` - maybe those types should just be in `@votingworks/types`? or we'd have to spin up a `@votingworks/api-client` lib?

there's possibly some crossover with what y'all are looking into WRT architecture changes?

Reply:
Yeah this feels like it will be exactly what we're gonna look at with the architecture changes. Though, as I said, I haven't worked with this code at all so your guess is as good as mine right now. I'll let @eventualbuddha shed some light

Reply:
Up to now I've been trying to follow some basic HTTP principles:
- use REST & HTTP methods to say what you want to happen
- use status codes to indicate success (200, arguably 201 but at least 200) and failure (400, 404, 405, etc)
- use content-type to indicate the content … type, and it should almost always be `application/json`

So for this PR I'd say _don't_ expect that error responses will have 200 status codes. In the future I'm thinking of abstracting away most of the HTTP layer with some sort of RPC setup, but we're not there yet and may never be.